src/datasets/iterable_dataset.py

```diff
@@ -3408,6 +3408,10 @@ def _concatenate_iterable_datasets(
     else:
         _check_column_names([col_name for dset in dsets for col_name in dset.features])
 
+    # Check format is consistent; if so, will set format for concatenated dataset
+    format_type_set = {dset._formatting.format_type for dset in dsets}
```

Comment:
`dset._formatting` doesn't always exist

So if `dset._formatting` is None for all the datasets, the formatting of the output dataset should be None as well

Reply:
I've added a check in the set comprehension, so that it skips elements where `dset._formatting` is `None`. This means that we will try to set the output format to that of the inputs who do have `dset._formatting`. Is that reasonable? or we should set the output format to `None` in this case?

Reply:
any discrepancy should result in formatting=None, if we want to be consistent with the `Dataset` API

Reply:
Sounds good! I've reworked the logic to account for that.